src/engine/sys/sdl_glimp.cpp

```diff
@@ -473,35 +467,121 @@ static void GLimp_DetectAvailableModes()
 		logger.Notice("Available modes: '%s'", buf );
 		ri.Cvar_Set( "r_availableModes", buf );
 	}
+
+	return true;
 }
 
-/*
-===============
-GLimp_SetMode
-===============
-*/
-static rserr_t GLimp_SetMode( int mode, bool fullscreen, bool noborder )
+enum class glProfile {
+	UNDEFINED = 0,
+	COMPATIBILITY = 1,
+	CORE = 2,
+};
+
+struct glConfiguration {
+	int major;
+	int minor;
+	glProfile profile;
+	int colorBits;
+};
+
+static bool operator!=(const glConfiguration& c1, const glConfiguration& c2) {
+	return c1.major != c2.major
+		|| c1.minor != c2.minor
+		|| c1.profile != c2.profile
+		|| c1.colorBits != c2.colorBits;
+}
+
+static const char* GLimp_getProfileName( glProfile profile )
 {
-	const char  *glstring;
-	int         perChannelColorBits;
-	int         alphaBits, depthBits, stencilBits;
-	int         samples;
-	int         i = 0;
-	SDL_Surface *icon = nullptr;
-	SDL_DisplayMode desktopMode;
-	Uint32      flags = SDL_WINDOW_SHOWN | SDL_WINDOW_OPENGL;
-	int         x, y;
-	GLenum      glewResult;
-	int         GLmajor, GLminor;
-	int         GLEWmajor, GLEWminor, GLEWmicro;
+	return profile == glProfile::CORE ? "core" : "compatibility";
```

Comment:
Now that I think about it, maybe I may add an ASSERT to make sure it's not undefined.

Reply:
@slipher I added an assert in your commit.